src/test/test-time-util.c

```diff
@@ -238,6 +238,11 @@ TEST(format_timespan) {
         test_format_timespan_accuracy(1);
         test_format_timespan_accuracy(USEC_PER_MSEC);
         test_format_timespan_accuracy(USEC_PER_SEC);
+
+        /* See issue #23928. */
+        _cleanup_free_ char *buf;
```

Comment:
Why not just `char buf[5]`?

Reply:
I do not know why, but valgrind does not detect the buffer-over-run if `buf[5]` is used.
ASAN may detect the issue even with `buf[5]`. But this is just a test code, hence I hope that's no problem.

Reply:
> I do not know why, but valgrind does not detect the buffer-over-run if `buf[5]` is used.

Because valgrind's `memcheck` doesn't check stack allocations.

Reply:
Ah, thanks.